FIX-#3624: Reformat modin readthedocs theme to look closer to pandas.

jeffreykennethli opened the pull request:
## What do these changes do?

* Change the readthedocs theme to be pydata-sphinx-theme (the same as [pandas](https://pandas.pydata.org/docs/index.html))
* Add a modin favicon
* Reformat the TOC tree to be narrower and deeper to better fit the pydata-sphinx-theme. The pydata-sphinx-theme takes the top-level documents and puts them on the top navbar, and since Modin's readthedocs had a lot of top-level documents, the navbar would be rendered with too much information. Thus multiple documents/directories were combined to end up with 5 top-level documents: Getting Started, Supported APIs, Advanced Usage, Developer, and Contact.

To build the html files locally, install the requirements in requirements-doc.txt and run `sphinx-build -b html docs/ ~/some_test_directory` from modin/.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` <!-- you can amend your commit with a signature via `git commit -amend -s` -->
- [x] Resolves #3624? <!-- issue must be created for each patch -->
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date <!-- if you have added, renamed or removed files or directories please update the documentation accordingly -->

devin-petersohn commented:
I think you need to add the theme to the requirements-doc.txt:

![image](https://user-images.githubusercontent.com/10732128/142959659-c6e4a7dd-60c1-4093-b860-e1f1cc4fc4ea.png)

dorisjlee commented on docs/index.rst:
```diff
@@ -117,69 +126,6 @@ nature, you get a fast DataFrame at 1MB and 1TB+.
 you can find 'Getting started' guide, system architecture and internal implementation details docs
 and lots of other useful information.**
```
Renamed to "Developer" Section. "Getting started" is not part of the Developer section. Would be good to link to the pages here directly.

dorisjlee commented:
The updated TOC tree largely looks good. 
Minor comments: For the Contact page, it would be good to remove the explicit URL. Would be good to add the [slack link](https://modin.org/slack.html) also as a section.

jeffreykennethli commented:
> The updated TOC tree largely looks good. Minor comments: For the Contact page, it would be good to remove the explicit URL. Would be good to add the [slack link](https://modin.org/slack.html) also as a section.

Removed explicit URL from Contact page and added Slack section on the top. Also added icon links to github, PyPI, Slack signup, and Mailing List on the top navbar.

> Renamed to "Developer" Section. "Getting started" is not part of the Developer section. Would be good to link to the pages here directly.

Done.

> I think you need to add the theme to the requirements-doc.txt:

Done.

dorisjlee commented on docs/contact.rst:
```diff
@@ -1,23 +1,26 @@
+-----
+
+Join the `Modin Slack`_ to join the conversation around making Modin better,
+ask questions to the community, and meet other Modin users.
```
```suggestion
discuss, and ask questions about all things Modin-related.
```

dorisjlee commented on docs/contact.rst:
```diff
@@ -1,23 +1,26 @@
+Slack
+-----
+
+Join the `Modin Slack`_ to join the conversation around making Modin better,
```
```suggestion
Join our `Slack`_ community to connect with Modin users and contributors,
```

dorisjlee commented on docs/contact.rst:
```diff
@@ -1,23 +1,26 @@
+Bug reports and feature requests can be directed to the issues_ page of the Modin
 GitHub repo.
 
+.. _Modin slack: https://modin.org/slack.html
```
```suggestion
.. _slack: https://modin.org/slack.html
```

vnlitvinov commented:
Here's the link to preview build: https://modin--3726.org.readthedocs.build/en/3726/

YarShev commented on docs/developer/index.rst:
```diff
@@ -0,0 +1,19 @@
+    :titlesonly:
+    
+    contributing
+    architecture
```
I see the following drop-down list for System Architecture clause ([link](https://modin--3726.org.readthedocs.build/en/3726/developer/index.html#developer)):
```
Modin PandasDataframe Objects
Generic Ray-based members
PandasOnRay Dataframe implementation
cuDFOnRay Dataframe Implementation
PandasOnDask Dataframe implementation
Experimental
Storage Formats
Query Compiler
PandasOnPython Dataframe implementation
BasePandasDataset
DataFrame Module Overview
Series Module Overview
```
I would prefer to see more high-level components (titles) there.

jeffreykennethli replied:
I set the option to show a depth of 4 to increase the visibility of the components in the System Architecture section. Unfortunately that also increases the depth of the other sections, but I think that's okay since the System Architecture section has the most components anyway.

YarShev commented on docs/getting_started/index.rst:
```diff
@@ -1,5 +1,19 @@
-Installation
-============
+===============
```
Why do we need this line over `Getting Started` section?

jeffreykennethli replied:
We don't need this. Removed.

YarShev commented on docs/supported_apis/defaulting_to_pandas.rst:
```diff
@@ -0,0 +1,34 @@
+Defaulting to pandas
+=======================================
```
```suggestion
Defaulting to pandas
===================
```

YarShev commented on docs/supported_apis/defaulting_to_pandas.rst:
```diff
@@ -0,0 +1,34 @@
+Defaulting to pandas
+=======================================
+
+.. _defaulting-to-pandas-mechanism:
```
What is this line needed for?

jeffreykennethli replied:
I don't believe this line .. _defaulting-to-pandas-mechanism: does anything. Removed.

YarShev commented on docs/supported_apis/defaulting_to_pandas.rst:
```diff
@@ -0,0 +1,34 @@
+has completed, we convert the DataFrame back into a partitioned Modin DataFrame. This
+way, operations performed after something defaults to pandas will be optimized with
+Modin.
+
```
```suggestion
```

jeffreykennethli replied:
Done.

jeffreykennethli commented on docs/contact.rst:
```diff
@@ -1,23 +1,26 @@
-https://github.com/modin-project/modin/issues
-
-Bug reports and feature requests should be directed to the issues_ page of the Modin
+Bug reports and feature requests can be directed to the issues_ page of the Modin
```
Added a section linking to our Discourse forum

jeffreykennethli commented on docs/conf.py:
```diff
@@ -135,8 +159,8 @@ def noop_decorator(*args, **kwargs):
-html_sidebars = {
-    "**": ["globaltoc.html", "relations.html", "sourcelink.html", "searchbox.html"]
-}
+# html_sidebars = {
```
I commented this out because the default html_sidebars for pydata_sphinx_theme is sufficient. See [docs](https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/configuring.html#configure-the-sidebar). Another way of saying it is these html_sidebars don't look good on the new theme. I've removed the html_sidebars comment for clarity.

vnlitvinov commented on docs/conf.py:
```diff
@@ -125,6 +130,25 @@ def noop_decorator(*args, **kwargs):
+            "url": "https://groups.google.com/forum/#!forum/modin-dev",
+            "icon": "fas fa-envelope-square",
+        },
+    ],
```
shouldn't discourse and conda-forge also be here?

jeffreykennethli replied:
Added discource and conda-forge to the icon-links section. Unfortunately icon-links only support [Font Awesome](https://fontawesome.com/) icons. I couldn't find an exact match for anaconda, so it's currently just a circle with a break. There's not a good workaround for this.

vnlitvinov commented on docs/developer/architecture.rst:
```diff
@@ -112,18 +112,8 @@ are also plans to expose the Modin DataFrame API as a reduced API set that encom
 the entire pandas/dataframe API. See `experimental features`_ for more information.
```
given you've renamed the section this link is now broken

jeffreykennethli replied:
Fixed the link and renamed to advanced features, thanks!

vnlitvinov commented on docs/developer/troubleshooting.rst:
```diff
@@ -1,198 +1,198 @@
-Troubleshooting
-===============
-
-We hope your experience with Modin is bug-free, but there are some quirks about Modin
```
can you make it so this file is not in a "whole thing changed" state? it's very hard to review.

jeffreykennethli replied:
It should be in a state of being moved, not "whole thing changed" state. Sorry if an old version was displayed as such.

vnlitvinov commented on docs/examples/index.rst:
```diff
@@ -1,13 +0,0 @@
-Examples
```
do we have this as a subsection anywhere now, or is this information gone altogether?

jeffreykennethli replied:
Thank you, I meant to add this to the bottom of the "Using Modin" Section.

vnlitvinov commented on docs/getting_started/using_modin.rst:
```diff
@@ -173,16 +174,10 @@ Specifying ``num_cpus`` limits the number of processors that Modin uses. You may
-.. _`blog post`: https://rise.cs.berkeley.edu/blog/pandas-on-ray-early-lessons/
-.. _`Jupyter Notebook`: https://gist.github.com/devin-petersohn/f424d9fb5579a96507c709a36d487f24#file-pandas_on_ray_blog_post_0-ipynb
 .. _`tools and features`: comparisons/upcoming_features.html
+.. _sklearn LinearRegression: https://github.com/modin-project/modin/blob/master/examples/modin-scikit-learn-example.ipynb
```
where is the link to this entry?

jeffreykennethli replied:
Moved over the Examples section to use this entry.

vnlitvinov commented on docs/index.rst:
```diff
@@ -113,73 +122,12 @@ nature, you get a fast DataFrame at 1MB and 1TB+.
 
 **Modin is currently under active development. Requests and contributions are welcome!**
 
-**If you are interested in contributions please refer to 'developer documentation' section, where
```
I feel it a little confusing when I open the main index page, and there's nothing on the left to navigate - I only have a "this page contains" navbar on the right, and I have to notice the upper navbar which is something I'm not used to.

Can we duplicate the upper menu here on the left then?

jeffreykennethli replied:
That's a good observation. The way pandas readthedocs gets around this is by having big icons for its four navbar sections. I think this would be better added in a following pull request, perhaps implemented similarly to how pandas does it.

vnlitvinov commented on docs/supported_apis/index.rst:
```diff
@@ -1,49 +1,30 @@
 
 For your convenience, we have compiled a list of currently implemented APIs and methods
 available in Modin. This documentation is updated as new methods and APIs are merged
 into the master branch, and not necessarily correct as of the most recent release. In
```
we can point here to a `stable` version of ReadTheDocs for Modin, which refers to the latest release.

jeffreykennethli replied:
Done.

vnlitvinov reviewed (approved):
LGTM, thanks @jeffreykennethli!

YarShev reviewed (approved):
LGTM

devin-petersohn reviewed (approved):
LGTM, thanks @jeffreykennethli!
